plugin-developer-learning-pathway/07-Custom-Database-Tables/01-Creating-and-Managing-Custom-Tables.md

```diff
@@ -16,54 +18,76 @@ Even still, there are times when creating a custom table can be more effective.
 
-You can access it in PHP like this:
+Because the `$wpdb` object exists in the global namespace, you can access it in PHP like this:
 
 ```php
 global $wpdb;
 ```
 
-The `$wpdb` instance also contains crucial information about the WordPress database which we'll need to properly create and manage our own custom tables. Let's see how as we create our first one.
+The `$wpdb` instance also contains crucial information about the WordPress database which you'll need to properly create and manage our own custom tables. Let's see how as we create our first one.
 
 ## Creating a Custom Table
 
-To start, register an activation hook for your plugin to create the custom table. Using this hook ensures the table immediately exists once your plugin is activated so that any following code can safely query against it.
+To start, create a new plugin to store the custom table code. 
+
+```
+<?php
+/**
+ * Plugin Name: Form submissions
+ * Version:     1.0.0
+ * Author:      Learn WordPress
+ * Author URI:  https: //learn.wordpress.org/
+ *
+ * @package wp-learn-form-submissions
+ */
+ ```
+
+Then, register an activation hook for your plugin to create the custom table. Using this hook ensures the table immediately exists once your plugin is activated so that any following code can safely query against it.
 
-When choosing a name for the custom table, be sure to use the configured table prefix by prepending the table name with the `$wpdb->prefix` variable. Also, use `$wpdb->get_charset_collate()` to ensure the custom table's collation matches the rest of the tables in WordPress for optimum data integrity, compatibility, and query performance.
+When choosing a name for the custom table, be sure to use the configured table prefix by prepending the table name with the `$wpdb->prefix` object property. Also, use `$wpdb->get_charset_collate()` method to ensure the custom table's collation matches the rest of the tables in WordPress for optimum data integrity, compatibility, and query performance.
 
-To create a custom table, use the `dbDelta()` function included in WordPress. This function is powerful because it can handle both table creation and future updates to the table's structure.
+To create a custom table, use the `dbDelta()` function included in WordPress. This function handles both table creation and future updates to the table's structure.
 
 ```php
-register_activation_hook( __FILE__, 'my_plugin_create_table' );
-function my_plugin_create_table() {
+register_activation_hook( __FILE__, 'wp_learn_custom_table_create_submissions_table' );
+function wp_learn_custom_table_create_submissions_table() {
 	global $wpdb;
 	$charset_collate = $wpdb->get_charset_collate();
 
-	$sql = "CREATE TABLE {$wpdb->prefix}my_plugin_table (
+	$sql = "CREATE TABLE {$wpdb->prefix}submissions (
```

Comment:
This makes way more sense, thank you! I was going to prefix everything with `wp_learn_*` but then realized the table would be named `wp_wp_learn_custom_table` which looked pretty silly. 🙂 To that point, is it okay to not illustrate the best practice of using a vendor prefix for custom table names?

Reply:
```
To that point, is it okay to not illustrate the best practice of using a vendor prefix for custom table names?
```

That's a good question. In this case, I consider `wp_learn` to be a valid vendor prefix, as the content and code for all the lessons "belong" to Learn WordPress...

Reply:
That's what I thought, too, but then I changed to using `my_plugin_` because I was thinking of how silly it will look to screenshare the `wp_wp_learn_submissions` table in the database. I also feel like it then adds more confusion/distraction because then I'd want to point out in the video that the _first_ `wp_` is the default table prefix in WordPress. 🫠 

Should we just leave it without a vendor prefix and not even address this detail? Because it's true: I didn't comment on best practice for naming custom tables... Should we?

Reply:
Personally, I think it's ok for this lesson. 

What I would do is add a "Best practices" section to the [following lesson](https://github.com/WordPress/Learn/issues/2721), to include things like this. 

You could also include something on how it's possible to use CREATE TABLE IF EXISTS and some of the other things @ryanwelcher mentioned in the initial review of the initial database tutorial (conversation here: https://github.com/WordPress/Learn/issues/1867#issuecomment-1743611499)